File: safety/snip/TestMakeAppenderMutate.go
func TestMakeAppenderMutate(t *testing.T) {
	t.Skip()
	suffix := []int{3, 4}
	append34 := MakeAppender(suffix)
	want := []int{1, 2, 3, 4}
	check := func() {
		if got := append34([]int{1, 2}); !reflect.DeepEqual(got, want) {
			t.Errorf("append34([]int{1, 2}) = %v; want %v", got, want)
		}
	}
	check()       // OK
	suffix[0] = 5 // Backdoor mutation of shared state
	check()       // FAIL
}

File: safety/snip/test_make_appender_mutate.rs
fn test_make_appender_mutate() {
    let mut suffix = [3, 4];
    let append34 = make_appender(&suffix);
    assert_eq!(vec![1, 2, 3, 4], append34(vec![1, 2]));
    suffix[0] = 5; // Won't compile.
    assert_eq!(vec![1, 2, 3, 4], append34(vec![1, 2]));
}
